runtime/src/main/cpp/Arrays.cpp

```diff
@@ -32,6 +38,16 @@ ALWAYS_INLINE inline void mutabilityCheck(KConstRef thiz) {
   }
 }
 
+template<typename T>
+inline void fillImpl(KRef thiz, KInt fromIndex, KInt toIndex, T value) {
+  ArrayHeader* array = thiz->array();
+  checkRangeIndexes(fromIndex, toIndex, array->count_);
+  mutabilityCheck(thiz);
+  for (KInt index = fromIndex; index < toIndex; ++index) {
+    *PrimitiveArrayAddressOfElementAt<T>(array, index) = value;
```

Comment:
Maybe sliding pointer instead?
I.e.
```
void fillImpl(KRef thiz, KInt fromIndex, KInt toIndex, T value) {
  ArrayHeader* array = thiz->array();
  checkRangeIndexes(fromIndex, toIndex, array->count_);
  mutabilityCheck(thiz);
  T* address = PrimitiveArrayAddressOfElementAt<T>(array, index);
  for (KInt index = fromIndex; index < toIndex; ++index) {
    *address++ = value;
  }
}```

Reply:
Just copied that from `IntArray.fill` implementation. I will try pointer increment instead as you're proposing.